File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthSmallest = function(root, k) {

    current = root;
    ans=[];

    function traverse(root){
        if(root == null) return;
        traverse(root.left);
        ans.push(root.val);
        traverse(root.right);
    }

    traverse(current);

    return ans[k-1];
    
};

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.java
class Solution {
    List<Integer> val = new ArrayList<Integer>();
    public int kthSmallest(TreeNode root, int k) {
        traverse(root);
        return val.get(k-1);
    }

    public void traverse(TreeNode root){
        if(root==null) return;

        traverse(root.left);
        val.add(root.val);
        traverse(root.right);
    }
}